internal/gcs/container.go

```diff
@@ -225,20 +245,83 @@ func (c *Container) Terminate(ctx context.Context) (err error) {
 // Wait waits for the container to terminate (or Close to be called, or the
 // guest connection to terminate).
 func (c *Container) Wait() error {
+	<-c.waitBlock
+	return c.waitError
+}
```

Comment:
Given the description of this function, wouldn't we want to wait on c.closeCh too?

Reply:
I wanted `waitBackground` to handle all the exit criteria, so that if the signaling from the guest connection are changed, `Wait` does not have to worry about it. Also, that way if some additional cleanup is needed between when the container is closed on the guest, and host goroutines are notified, then `waitBackground` (and `Close`) can take care of them and then close `waitBlock` without having to worry about race conditions